src/snowflake/connector/auth/_auth.py

```diff
@@ -166,7 +168,7 @@ def authenticate(
         mfa_callback: Callable[[], None] | None = None,
         password_callback: Callable[[], str] | None = None,
         session_parameters: dict[Any, Any] | None = None,
-        timeout: int = 120,
+        mfa_timeout: int | None = None,  # max time waiting for MFA response
```

Comment:
`Auth.authenticate` does not seem to be exposed to the public. The documentation shows that [authentication is only done through the connector](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-api). Furthermore, even internally, the `timeout` argument to `Auth.authenticate` is never used.

I changed the name because this argument only affects the `join` timeout while waiting for MFA authentication, so I found `timeout` to be a strange name as it implies something more general.